### ISSUE TITLE
Wrapped tf.nn.RNNCell* layers are incompatible with tf.keras.layers.Bidirectional

### ISSUE THREAD

**Issue by guillaumekln** [tensorflow/tensorflow#26581]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Ubuntu 16.04
- TensorFlow installed: binary
- TensorFlow version: `2.0.0.dev20190311`
- Python version: 3.6.6

**Describe the current behavior**

In the TensorFlow 2.0 preview, the `tf.nn.RNNCellDropoutWrapper` and `tf.nn.RNNCellResidualWrapper` wrappers are incompatible with `tf.keras.layers.Bidirectional`. It raises an "Unknown layer" exception.

**Describe the expected behavior**

The RNN classes in `tf.nn` should be compatible with all Keras RNN layers.

**Code to reproduce the issue**

```python
import tensorflow as tf

cell = tf.keras.layers.LSTMCell(10)
cell = tf.nn.RNNCellResidualWrapper(cell)

rnn = tf.keras.layers.RNN(cell)
rnn = tf.keras.layers.Bidirectional(rnn)
```

**Other info / logs**

```text
Traceback (most recent call last):
  File "test/residual_wrapper.py", line 30, in <module>
    rnn = tf.keras.layers.Bidirectional(rnn)
  File "/lib/python3.6/site-packages/tensorflow/python/keras/layers/wrappers.py", line 393, in __init__
    self.backward_layer = layer.__class__.from_config(config)
  File "/lib/python3.6/site-packages/tensorflow/python/keras/layers/recurrent.py", line 920, in from_config
    cell = deserialize_layer(config.pop('cell'), custom_objects=custom_objects)
  File "/lib/python3.6/site-packages/tensorflow/python/keras/layers/serialization.py", line 95, in deserialize
    printable_module_name='layer')
  File "/lib/python3.6/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 181, in deserialize_keras_object
    config, module_objects, custom_objects, printable_module_name)
  File "/lib/python3.6/site-packages/tensorflow/python/keras/utils/generic_utils.py", line 166, in class_and_config_for_serialized_keras_object
    raise ValueError('Unknown ' + printable_module_name + ': ' + class_name)
ValueError: Unknown layer: ResidualWrapperV2
```

@qlzh727

**Comment by qlzh727** [comment]
Thanks for reporting the issue, I will send some fix within this week.

**Comment by ruanchong** [comment]
This is weird. Why `tf.nn.rnn_cell.*` is moved to `tf.keras`, but ResidualWrapper and DeviceWrapper are not (and they are inheriting new base classes such like `ResidualWrapperV2`)?

**Comment by qlzh727** [comment]
Sorry for the very late reply. The issue was caused by the serial/deserialization for a non-keras object, and the nn.RNNCell wrappers are not keras object. The commit https://github.com/tensorflow/tensorflow/commit/0667725ba4e010c06fcf5e2a388eb8c2f3f6efe1 is trying to fix the custom object serialization issue, which allows you to do something like https://gist.github.com/fchollet/9e361d1fe9046b512a2f3e0d31382a96.

We are still discussing the issue internally and see how we will fix this issue.

**Comment by qlzh727** [comment]
Sorry for the long wait, this should now by fixed by e62dc433fcce833313b4174e20fc24c418593d27
